### PR TITLE
Re-Addressing New Addressing Class

### PR DESCRIPTION
**Pull request by adamgering**
First of a few re-addressing PRs, this creates the
new addressing class.

### CODE REVIEW

File: rbac/common/addresser/task.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.admin = TaskAdminAddress()
+
+    @property
+    def address_type(self):
+        return AddressSpace.TASKS_ATTRIBUTES
+
+    def address(self, object_id, target_id=None):
+        return legacy.make_task_attributes_address(task_id=object_id)
+
+    def address_is(self, address):
```
[the remark this answers is not in the document]
Comment (adamgering) [reply rbac/common/addresser/task.py]: The only function that has some duplication is address_is. That duplication is temporary, thus it is not implemented in the base class.

[the remark this answers is not in the document]
Comment (adamgering) [reply rbac/common/addresser/task.py]: Tasks will likely disappear, they are just roles without members; but this PR is a refactor not a design change.

File: rbac/addressing/addresser.py [hunk 2]
```diff
@@ -75,8 +75,8 @@ class TaskRelationshipNamespace(enum.IntEnum):
 
 FAMILY_NAME = "rbac"
 FAMILY_VERSION = "1.0"
-NS = sha512(FAMILY_NAME.encode()).hexdigest()[:6]
-FAMILY_PREFIX = "9f4448"
+NAMESPACE = sha512(FAMILY_NAME.encode()).hexdigest()[:6]
+NS = NAMESPACE
```
Comment (kthblmfld) [inline rbac/addressing/addresser.py]: Can NS go away now?

File: rbac/common/proposal/proposal_state.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        """Make an address for the given user_id"""
+        return addresser.make_proposal_address(
+            object_id=object_id, related_id=target_id
+        )
+
+    def make(self, object_id):
+        """Makes the state storage object"""
+        raise NotImplementedError("Class must implement this method")
+
+    def create(self, signer_keypair, message, object_id=None, target_id=None):
```
Comment (nadiabahrami) [inline rbac/common/proposal/proposal_state.py]: This appears to be duplicated from base state and can be removed

File: rbac/common/proposal/proposal_state.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    def state_proto(self):
+        return protobuf.proposal_state_pb2.Proposal
+
+    def address(self, object_id, target_id=None):
+        """Make an address for the given user_id"""
+        return addresser.make_proposal_address(
+            object_id=object_id, related_id=target_id
+        )
+
+    def make(self, object_id):
```
Comment (nadiabahrami) [inline rbac/common/proposal/proposal_state.py]: Appears to be implemented in basestate and can be removed

File: tests/rbac/common/addresser/address_assertions.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+ADDRESS_CLASS_METHODS = ["address", "address_is", "hash", "unique_id"]
+ADDRESS_CLASS_PROPS = ["address_type"]
+
+
+class AddressAssertions(CommonAssertions):
+    def __init__(self, *args, **kwargs):
+        CommonAssertions.__init__(self, *args, **kwargs)
+
+    def assertIsAddressClass(self, value):
```
Comment (nadiabahrami) [inline tests/rbac/common/addresser/address_assertions.py]: docstrings on what is being tested

File: rbac/common/manager/base_state.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    @property
+    def state_proto(self):
+        raise NotImplementedError("Class must implement this method")
+
+    @property
+    def message_fields_not_in_state(self):
+        """Fields that are on the message but not stored on the state object"""
+        return []
+
+    def address(self, object_id, target_id):
```
Comment (nadiabahrami) [inline rbac/common/manager/base_state.py]: should =None be default for target_id

File: rbac/addressing/addresser.py (hunk 2 above)
Comment (adamgering) [reply rbac/addressing/addresser.py]: rbac/addressing/addresser.py
Disappears in the next PR

[the remark this answers is not in the document]
Comment (adamgering) [reply rbac/addressing/addresser.py]: rbac/addressing/addresser.py
Disappears in the next PR

File: rbac/common/addresser/app.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+from hashlib import sha512
+import re
+
+
+FAMILY_NAME = "rbac"
```
[the remark this answers is not in the document]
Comment (adamgering) [reply rbac/common/addresser/app.py]: rbac/addressing/addresser.py
Disappears in the next PR

File: rbac/common/proposal/proposal_state.py (hunk 4 above)
Comment (adamgering) [reply rbac/common/proposal/proposal_state.py]: what line are you commenting on?

File: rbac/common/proposal/proposal_state.py (hunk 3 above)
Comment (adamgering) [reply rbac/common/proposal/proposal_state.py]: what line are you commenting on?

Comment (nadiabahrami) [reply rbac/common/proposal/proposal_state.py]: The create method

File: rbac/common/proposal/proposal_state.py (hunk 4 above)
Comment (nadiabahrami) [reply rbac/common/proposal/proposal_state.py]: The make function

File: tests/rbac/common/addresser/address_assertions.py (hunk 5 above)
Comment (adamgering) [reply tests/rbac/common/addresser/address_assertions.py]: resolved

**Pull request merged**